### ISSUE TITLE
Add keyboard shortcuts for updating the properties

### ISSUE THREAD

**Issue by lipis** [excalidraw/excalidraw#2649]
We already have shortcuts for the dropdowns in colors and arrowheads, but let's take it to another step and add a few more shortcuts for updating the properties without a mouse.

It was initially suggested in #1104 by @vullankib, where you can find more details.

It can be done in steps, and this issue is here to cover all cases. It will be updated as we go.

|                    | Property         | Shortcut | Description                                                                       |
|--------------------|------------------|:--------:|-----------------------------------------------------------------------------------|
| :heavy_check_mark: | Stroke           |   `s`    | Open the color picker for stroke                                                  |
| :heavy_check_mark: | Background       |   `g`    | Open the color picker for background                                              |
|                    | Stroke width     |          | Change the stroke width to the next one                                           |
|                    | Stroke style     |          | Change the stroke style to the next one                                           |
|                    | Sloppiness       |          | Change the sloppiness to the next one                                             |
|                    | Edges            |          | Toggle between two options                                                        |
|                    | Arrowheads start |          | Open the dropdown for the arrowheads start                                        |
|                    | Arrowheads end   |          | Open the dropdown for the arrowheads end                                          |
|                    | Font size        |          | Change the font size to the next one                                              |
|                    | Font family      |          | Change the font family to the next one                                            |
|                    | Text align       |          | Change the text align to the next one                                             |
|                    | Opacity          |          | Focus on the opacity slider and using `Left`/`Right` it should change the opacity |

### Tasks

- [ ] Find best suitable shortcuts for each property
- [ ] Implement shortcuts (can be done in multiple PRs)
- [ ] Add shortcuts to shortcuts dialog
- [ ] Maybe show shortcuts next to the label (where applicable)

**Comment by dwelle** [comment]
If we're gonna go with cycling as well, I'd not use custom letters and instead use modifiers (e.g. `shift+g`).

We also need to figure out conflicts with other apps and standards.

- I'd drop `Canvas background` from the list for now. Not that useful.
- `c` is used for comments in Figma. I'd keep that shortcut reserved even if we don't have immediate plans for it.
- For `fill color` I'd go either with `g` (used in desktop editors), `b` (background), or `f` (though `f` is used by figma for Frame). Preference on `g`.
- For `stroke color` I'd go with `s` (stroke) or `f`. Maybe `d`.

**Comment by JakubKoralewski** [comment]
@dwelle Why the `s` and `g` shortcut doesn't work to change the selected color while a tool is active and nothing is selected? Instead pressing `s` goes back to the Selection tool and pressing `g` does nothing.

**Comment by dwelle** [comment]
Oh, I believe `s` is a legacy shortcut for the selection tool (no longer documented). I'd say we can now remove it.

Good catch about it not working if nothing is selected. Though how should it work when using the `selection `tool? One way would be to switch to the first non-selection tool, in this case the `rectangle`.

**Comment by JakubKoralewski** [comment]
> Though how should it work when using the selection tool?

I'd expect it do nothing, really. At best its a misclick if someone pressed it cause it doesn't really make sense. Unless maybe if someone pressed `g` then it would make sense to change the background color?

> Wanna take this up?

Ok I got it to work, so yeah I'd like to. Just don't know where to start.

**Comment by artjomsR** [comment]
Why was this issue closed? OP originally talked about having shortcuts for all the functionality in the formatting window but it seems only `s` and `g` shortcuts got implemented / fixed?

**Comment by Cito** [comment]
Yes, please reopen this, since the other properties still have no shortcuts.

**Comment by Frazer** [comment]
Text align could use the existing align shortcuts.  IE:  when text is selected, using this shortcuts aligns the text, especially as they dont do anything when only 1 element is selected any way.
Align top -> Cmd Shift Up
Align bottom ->  Cmd Shift Down
Align left -> Cmd Shift Left
Align right -> Cmd Shift Right

**Comment by kyle-tennison** [comment]
Would still like something for stroke width... it doesn't seem implemented yet either

**Comment by miroslavzeman** [comment]
+1 🙏
